### ISSUE TITLE
Exclude MA0042 for sync redis call with `CommandFlags.FireAndForget` flag

### ISSUE THREAD

**Issue by nesterenko-kv** [meziantou/Meziantou.Analyzer#457]
I expect this rule to be not applied to this code:
```cs
using System;
using System.Threading.Tasks;
using StackExchange.Redis;

class Sample
{
    private readonly IDatabase _redisDatabase;

    public Sample(
        IDatabase redisDatabase
        )
    {
        _redisDatabase = redisDatabase;
    }

    async Task A()
    {
        var _ = _redisDatabase.StringSet("""", ""value"", TimeSpan.FromDays(1), When.Always, CommandFlags.FireAndForget);
    }
}
```

This applies to all methods with `CommandFlags.FireAndForget` being applied.

**Comment by meziantou** [comment]
I'm not an expert of StackExchange.Redis, so maybe the following is wrong. I just quickly looked at the code to understand if async is useful.

Even if you use `CommandFlags.FireAndForget` the async method seems useful. For instance, `StringSet` ends up calling `WriteMessageTakingWriteLockSync`. This method may use `SemaphoreSlim.Wait` instead of `SemaphoreSlim.WaitAsync`. So, it seems you should use the async equivalent to avoid blocking the ThreadPool.

**Comment by nesterenko-kv** [comment]
As far as I can see, in `WriteMessageTakingWriteLockSync` there's `TryPushToBacklog` method which is pushing messages to `ConcurrentQueue<Message> _backlog` and `StartBacklogProcessor` is proccesing them in the background.

**Comment by meziantou** [comment]
Is there a reason for `TryPushToBacklog` to always return `true`? Because if it can return `false`, then you end up in the lock and the async path is pertinent.

**Comment by nesterenko-kv** [comment]
It's funny, the deeper you climb into the jungle, the more difficult it is to understand what is happening there at all. There is a suspicion that such a situation is possible... All of the [examples](https://stackexchange.github.io/StackExchange.Redis/Basics.html#sync-vs-async-vs-fire-and-forget) in GH Wiki for StackExchange.Redis use sync method with FireAndForget flag...
